tracer.go

```diff
@@ -73,6 +73,12 @@ type Options struct {
 	// When set, it attempts to exacerbate issues emanating from use of Spans
 	// after calling Finish by running additional assertions.
 	DebugAssertUseAfterFinish bool
+	// DisableSpanPool disables reuse of the underlying span structs.
+	// Buyer Beware, any operations called on a span after Finish is undefined
+	// behavior. The main usecase of disabling the pool is to allow the creation
```

Comment:
I'd feel better if:
- SpanPool were off by default, and a new option 'EnableSpanPool' had to be set to use it
- Span use-after-finish were permitted except when EnableSpanPool is set
- DebugAssertUseAfterFinish disappears (it's implied by EnableSpanPool)